server/app/services/scheduler/filter/cpu.rb

```diff
@@ -0,0 +1,25 @@
+module Scheduler
+  module Filter
+    class Cpu
+
+      ##
+      # @param [GridService] service
+      # @param [Integer] instance_number
+      # @param [Array<HostNode>] nodes
+      # @return [Array<HostNode>]
+      def for_service(service, instance_number, nodes)
+        return nodes if service.cpus.nil?
+
+        candidates = nodes.dup.select { |n|
+          n.cpus.to_f >= service.cpus.to_f
+        }
+
+        if candidates.empty?
+          raise Scheduler::Error, "Did not find any nodes with #{service.cpus} CPUs available"
```

Comment:
The filtering logic is also a little dubious... `--cpus 4` doesn't imply that the service needs at least four CPUs to run... there's no reservation of CPU resources that would imply that the CPUs are somehow "available" for the service.

If the host node has other CPU-hungry services scheduled, then it won't get it's four CPUs anyways.

Reply:
>  guaranteed to be able to access

Reply:
What do you mean? If you're talking about the [`docker run`](https://docs.docker.com/engine/admin/resource_constraints/#configure-the-default-cfs-scheduler) docs, then your quote is very much out of context:

> the container will be guaranteed to be able to access ***at most*** one and a half of the CPUs